src/runtime_src/core/pcie/driver/linux/xocl/subdev/xgq.c

```diff
@@ -1677,6 +1680,111 @@ static int vmr_status_query(struct platform_device *pdev)
 	return vmr_control_op(pdev, XGQ_CMD_VMR_QUERY);
 }
 
+static void clk_scaling_cq_result_copy(struct xocl_xgq_vmr *xgq,
+                                       struct xocl_xgq_vmr_cmd *cmd)
+{
+	struct xgq_cmd_cq_default_payload *payload =
+		(struct xgq_cmd_cq_default_payload *)&cmd->xgq_cmd_cq_payload;
+	struct xgq_cmd_cq_clk_scaling_payload *cs_payload=
+		(struct xgq_cmd_cq_clk_scaling_payload *)&xgq->xgq_cq_payload;
+
+	mutex_lock(&xgq->xgq_lock);
+	memcpy(&xgq->xgq_cq_payload, payload, sizeof(*payload));
+	xgq->pwr_scaling_ovrd_limit = cs_payload->pwr_scaling_limit;
+	xgq->temp_scaling_ovrd_limit = cs_payload->temp_scaling_limit;
+	mutex_unlock(&xgq->xgq_lock);
+}
+
+static int clk_scaling_configure_op(struct platform_device *pdev,
+                                    enum xgq_cmd_clk_scaling_app_id aid,
+                                    bool enable, uint16_t pwr_ovrd_limit,
+                                    uint8_t temp_ovrd_limit)
+{
+	struct xocl_xgq_vmr *xgq = platform_get_drvdata(pdev);
+	struct xocl_xgq_vmr_cmd *cmd = NULL;
+	struct xgq_cmd_clk_scaling_payload *payload = NULL;
+	struct xgq_cmd_sq_hdr *hdr = NULL;
+	int ret = 0;
+	int id = 0;
+
+	if (xgq->xgq_halted) {
+		XGQ_WARN(xgq, "VMR XGQ service is haulted. skip.");
+		return -EIO;
+	}
+
+	cmd = kmalloc(sizeof(*cmd), GFP_KERNEL);
+	if (!cmd) {
+		XGQ_ERR(xgq, "kmalloc failed, retry");
+		return -ENOMEM;
+	}
+
+	memset(cmd, 0, sizeof(*cmd));
+	cmd->xgq_cmd_cb = xgq_complete_cb;
+	cmd->xgq_cmd_arg = cmd;
+	cmd->xgq_vmr = xgq;
+
+	payload = &(cmd->xgq_cmd_entry.clk_scaling_payload);
+	payload->aid = aid;
+	if (aid == XGQ_CMD_CLK_THROTTLING_AID_CONFIGURE)
+	{
+		payload->scaling_en = enable ? 1 : 0;
+		if (pwr_ovrd_limit)
+			payload->pwr_scaling_ovrd_limit = pwr_ovrd_limit;
+		if (temp_ovrd_limit)
+			payload->temp_scaling_ovrd_limit = temp_ovrd_limit;
+	}
+
+	hdr = &(cmd->xgq_cmd_entry.hdr);
+	hdr->opcode = XGQ_CMD_OP_CLK_THROTTLING;
+	hdr->state = XGQ_SQ_CMD_NEW;
+	hdr->count = sizeof(*payload);
```

Comment:
The `count `name and `sizeof(*payload`) don't seem to align.

One is normally referring to a 'number of elements'  and the other the size (bytes) to represent the items.  

For example, if one has:
`uint32_t myArray[10];`
The `count `(e.g., number of elements would be 10) and the `size `(e.g., number of bytes) would be 40.

If this code is truly what is desired, then a simply comment explaining why would greatly be helpful.

Reply:
Hi @rozumx thank you for the review. count represents the size in bytes only, copying it's definition from xgq_cmd_vmr.h file below.
 "@count:  how many data returned in bytes".